### DISCUSSION TITLE
Having trouble with capture output of ImageMagick in a bash script

### DISCUSSION THREAD

**Discussion by sunhasgothishaton** [ImageMagick/ImageMagick#5399] [Help] [answered]
i am using ImageMagick 7.1.0-44 AppImage on Ubuntu 20.04 LTS

I have a bash script to use imagemagick to generate an SSIM metric between two photos (magick compare).
I run the lines

SSIM=$(magick compare -metric SSIM -rotate '270<' 101-0133_STD.JPG '/home/rob/Pictures_to_sort/1a1_1b1/photos/Edinburgh/Stitched - Edinburgh/101-0133_STD.JPG' -compose src output/101-0133_STD.JPG.jpg)
echo "${SSIM} : 101-0133_STD.JPG"
echo "${SSIM} : 101-0133_STD.JPG" >> output/diff.log

ImageMagick works and prints the metric to the command line, rather than being saved to the bash variable SSIM. Printing ${SSIM} doesn't print anything as it doesn't appear in the log file (diff.log). I don't know if this is a bash thing, appimage thing or an imagemagick thing.

Can something help?

Thanks,
Rob

**Comment by snibgo** [comment, marked as answer]
For "`magick compare`", the text output is written to stderr, not stdout. You can redirect stderr to stdout like this:
```
$ COMP=$(magick compare in1.png in2.png -metric SSIM NULL: 2>&1 )
```

**Comment by sunhasgothishaton** [comment]
Fantastic, thank you. I wouldn't have thought of that.
